io/FV3GFS_io.F90

```diff
@@ -1547,6 +1551,36 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain, warm_sta
       enddo
     endif
 
+    if (sfc_var2(i,j,47) < -9990.0_r8) then
+      if (Model%me == Model%master ) call mpp_error(NOTE, 'gfs_driver::surface_props_input - computing snodi')
+!$omp parallel do default(shared) private(nb, ix, tem)
+      do nb = 1, Atm_block%nblks
+        do ix = 1, Atm_block%blksz(nb)
+          if (Sfcprop(nb)%fice(ix) > zero) then
+            tem = one / (Sfcprop(nb)%fice(ix)*(one-Sfcprop(nb)%landfrac(ix))+Sfcprop(nb)%landfrac(ix))
+            Sfcprop(nb)%snodi(ix)  = min(Sfcprop(nb)%snowd(ix) * tem, 3.0)
+          else
+            Sfcprop(nb)%snodi(ix)  = zero
+          endif
+        enddo
+      enddo
+    endif
+
+    if (sfc_var2(i,j,48) < -9990.0_r8) then
+      if (Model%me == Model%master ) call mpp_error(NOTE, 'gfs_driver::surface_props_input - computing weasdi')
+!$omp parallel do default(shared) private(nb, ix, tem)
+      do nb = 1, Atm_block%nblks
+        do ix = 1, Atm_block%blksz(nb)
+          if (Sfcprop(nb)%fice(ix) > zero) then
+            tem = one / (Sfcprop(nb)%fice(ix)*(one-Sfcprop(nb)%landfrac(ix))+Sfcprop(nb)%landfrac(ix))
+            Sfcprop(nb)%weasdi(ix)  = Sfcprop(nb)%weasd(ix)*tem
```

Comment:
I saw in GFS_typedefs.meta, the weasd is already defined as: "water equiv of acc snow depth over land and sea ice", why do we need to devide "tem" and not directly set weasdi=weasd?

Reply:
Jun,
In "GFS_surface_composites.F90" weasd is computed in the loop
"      if (frac_grid) then

        do i=1, im

          ! Three-way composites (fields from sfc_diff)
          txl   = landfrac(i)            ! land fraction
          wfrac = one - txl              ! ocean fraction
          txi   = cice(i) * wfrac        ! txi = ice fraction wrt whole cell
          txo   = max(zero, wfrac-txi)   ! txo = open water fraction

         !gflx(i)   = txl*gflx_lnd(i)  + txi*gflx_ice(i)  + txo*gflx_wat(i)
          ep1d(i)   = txl*ep1d_lnd(i)  + txi*ep1d_ice(i)  + txo*ep1d_wat(i)
          weasd(i)  = txl*weasd_lnd(i) + txi*weasd_ice(i)
          snowd(i)  = txl*snowd_lnd(i) + txi*snowd_ice(i)
         !tprcp(i)  = txl*tprcp_lnd(i) + txi*tprcp_ice(i) + txo*tprcp_wat(i)"

So, if you set weasd_lnd=weasd/(txi+txl) and wead_ice=weasd/(txi+txl) and then use the formula above you get the right answer. i.el.
txl*weasd_lnd(i) + txi*weasd_ice(i) = weasd(i) * (txl+txi) / (txl+txi) = weasd(i)

Please note that this has any impact when "txl+txi) is < 1.  If txl+txi=1, then there is no impact - so this affects only the fractional grid.

Reply:
Moorthi, I see the changes only have impact on fraction grid when txl+txi< 1. Just for discussion. I am wondering since in the code weasd/snowd are computed as composite fields GFS_surface_composites.F90 fields over land and ice, could we just use txl/(txl+txi) and txi/(txl+txi) in weasd/snowd computation?

Reply:
I am not sure I completely understand this.
For completeness,
weasd = txl * weasd_lnd + txi * weasd_ice + txo * weasd_ocn
and txl+txi+txo=1.0
However, weasd_ocn=0.0; so it is left out from the computation.
What is done in initial condition is going to be changed within one timestep by the snow imported from CICE6 when coupled.
Moorthi